pkg/storage/utils/filelocks/filelocks.go

```diff
@@ -140,8 +141,11 @@ func ReleaseLock(lock *flock.Flock) error {
 
 	err = lock.Unlock()
 	if err == nil {
-		if !lock.Locked() {
+		if !lock.Locked() && !lock.RLocked() {
 			err = os.Remove(n)
+			if err != nil && strings.Contains(err.Error(), "no such file or directory") {
```

Comment:
try `os.IsNotExist()`

Reply:
:+1: Thanks!

Reply:
Actually it should be `errors.Is(err, fs.ErrNotExist)` now because `os.IsNotExist()` is deprecated but this is fine for now.

Reply:
Fixed it quickly, pipeline needed restart anyways...